dev/tests/integration/testsuite/Magento/Review/Controller/CaseCheckUnsuccessfulAddingProductReviewTest.php

```diff
@@ -0,0 +1,91 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Review\Controller;
+
+use Magento\Catalog\Api\Data\ProductInterface;
+use Magento\Catalog\Api\ProductRepositoryInterface;
+use Magento\Framework\Data\Form\FormKey;
+use Magento\Framework\Message\MessageInterface;
+use Magento\TestFramework\Request;
+use Magento\TestFramework\TestCase\AbstractController;
+
+/**
+ * Test review product controller behavior
+ *
+ * @magentoAppArea frontend
+ */
+class CaseCheckUnsuccessfulAddingProductReviewTest extends AbstractController
+{
+    /**
+     * Test adding a review for allowed guests with incomplete data by a not logged in user
+     *
+     * @magentoDbIsolation enabled
+     * @magentoAppIsolation enabled
+     * @magentoConfigFixture default_store catalog/review/allow_guest 1
+     * @magentoDataFixture Magento/Catalog/_files/products.php
+     */
+    public function testAttemptForGuestToAddReviewsWithIncompleteData()
+    {
+        $product = $this->getProduct();
+        /** @var FormKey $formKey */
+        $formKey = $this->_objectManager->get(FormKey::class);
+        $post = [
+            'nickname' => 'Test nick',
+            'title' => 'Summary',
+            'form_key' => $formKey->getFormKey(),
+        ];
+        $this->prepareRequestData($post);
+        $this->dispatch('review/product/post/id/' . $product->getId());
+        $this->assertSessionMessages(
+            $this->equalTo(['Please enter a review.']),
+            MessageInterface::TYPE_ERROR
+        );
+    }
+
+    /**
+     * Test adding a review for not allowed guests by a guest
+     *
+     * @magentoDbIsolation enabled
+     * @magentoAppIsolation enabled
+     * @magentoConfigFixture default_store catalog/review/allow_guest 0
+     * @magentoDataFixture Magento/Catalog/_files/products.php
+     */
+    public function testAttemptForGuestToAddReview()
```

Comment:
In this test we need to make sure that checkout for guests is disabled in the configuration. Otherwise, in case if database cleanup is disabled and some other test changes this configuration value - we have another result. 
By setting the configuration value manually every time we will make this test more bulletproof.

Reply:
Now we'll have for sure the needed configs for our cases.
Thank you for suggestion.